frontend/src/modules/logs/EventType.tsx

```diff
@@ -63,27 +57,20 @@ export const getEventDetails = (event: LogEvent, eventType: EventType): string =
     }
     return '';
   } else if (event.type === LogEventType.system) {
-    if (event.code === LogEventCode.mcu_backend_connection_down) {
-      return 'Hardware controller lost connection';
-    }
-    if (event.code === LogEventCode.backend_mcu_connection_down) {
-      return 'Backend server lost connection from hardware controller';
-    }
-    if (event.code === LogEventCode.backend_frontend_connection_down) {
-      return 'User interface lost connection';
-    }
-    if (event.code === LogEventCode.frontend_backend_connection_down) {
-      return 'User interface lost connection';
-    }
-    if (event.code === LogEventCode.mcu_backend_connection_up) {
-      return 'Hardware controller connected';
-    }
-    if (event.code === LogEventCode.backend_frontend_connection_up) {
-      return 'User interface connected';
-    }
-    if (event.code === LogEventCode.charger_disconnected) {
-      return 'Battery charger is disconnected';
-    }
+    const eventDetailsMap: Record<number, string> = {
```

Comment:
`store/controller/types.ts` uses `Map` instead of `Object` (which you've type-annotated as `Record` here) - since all keys are known before the frontend is started, and the keys are enums rather than strings, maybe it makes sense to use `Map` here too? (this also applies to `EventTypeMap` below)
https://github.com/pez-globo/pufferfish-software/blob/0695a3e034d35ef18126f10fe8acd058c480ec40/frontend/src/store/controller/types.ts#L91

The MDN Web Docs page for `Map` has a useful table comparing the tradeoffs between `Map` and `Object: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map

Since `Map` exposes a `has` method, we can use a more clear/explicit check for a missing event code (`if (!eventDetailsMap.has(event.code))`) to return a default value, instead of using `||`.

Reply:
Switched to map, and using `eventDetailsMap.get(event.code)` since we need that anyway then handling the `undefined` by returning default value.